include/ck_tile/ops/gemm/pipeline/gemm_universal_pipeline_ag_bg_cr_policy.hpp

```diff
@@ -22,286 +22,128 @@ struct UniversalGemmPipelineAgBgCrPolicy
 
     static constexpr bool TransposeC = true;
 
+    template <typename Problem, typename DataType, index_t kMNPerBlock>
+    CK_TILE_HOST_DEVICE static constexpr auto GetKPack()
+    {
+        constexpr index_t kBlockSize  = Problem::kBlockSize;
+        constexpr index_t kKPerBlock  = Problem::BlockGemmShape::kK;
+        constexpr index_t kMaxVecLoad = 16 / sizeof(DataType);
+        constexpr index_t kMinVecLoad = 4 / sizeof(DataType);
+
+        constexpr index_t total_pixels = kMNPerBlock * kKPerBlock / kBlockSize;
+
+        constexpr index_t kVecLoad = ((total_pixels / kMaxVecLoad) >= kMinVecLoad)
+                                         ? kMaxVecLoad
+                                         : (total_pixels / kMinVecLoad);
```

Comment:
This equations is very confusing to me. `total_pixels / kMaxVecLoad) >= kMinVecLoad` this condition I believe would be almost always true, since the `total_pixels` which is a number of elements per thread (since you divide the tile size by the number of thread in workgroup) will be quite large and almost always a multiple of block size.  Additionally the `false` value `total_pixels / kMinVecLoad`  is again even more confusing - how would you like to use the number of vector load instructions as a vector load size??? Let's say that you have `kMaxVecLoad` equal to 4 and `kMinVecLoad` equal to 1, then you would have to have 12 elements in total (tile size) to get this false condition value and this will result in `kVecLoad` equal to 12... it doesn't make any sense for me.

Reply:
Fixed